code/modules/antagonists/changeling/headslug_eggs.dm

```diff
@@ -0,0 +1,44 @@
+#define EGG_INCUBATION_TIME (4 MINUTES)
+
+/// The actual organ that the changeling headslug implants into a dead host.
+/obj/item/organ/internal/body_egg/changeling_egg
+	name = "changeling egg"
+	desc = "Twitching and disgusting."
+	/// The mind of the original changeling that gave forth to the headslug mob.
+	var/datum/mind/origin
+	/// Tracks how long the egg has been growing.
+	var/time = 0
+
+/obj/item/organ/internal/body_egg/changeling_egg/egg_process(seconds_per_tick, times_fired)
+	// Changeling eggs grow in dead people
+	time += seconds_per_tick * 10
+	if(time >= EGG_INCUBATION_TIME)
+		pop()
+		Remove(owner)
+		qdel(src)
```

Comment:
I should've commented this earlier but shouldn't ``qdel`` be on pop in case it's ever called anywhere else in the future

Reply:
no because how are we going to call `Remove()` on a qdel'd item. `Remove()` is needed for this type of organ stuff. order of operations is critical here. if someone wants to piss around with extending this they should spin it out into another proc

Reply:
Then maybe ``Remove()`` should also be on ``pop()``?

Reply:
i don't really get why you want this to be a public facing proc when it hasn't been touched in the last five years and will likely not operate in that context again. this is a mob refactoring headslugs, this is just a file that i split out of the mob file to keep things organized. i could have easily done a PR four days ago to fix any/all of these issue with the eggs in its own pertinent PR such that it could be assessed on its own merits and actually get a proper lookover from the concerned organ people (since I have very very very sparse knowledge working with organs), rather than be completely ignored since this PR is billed as a basic mob refactor.

if you don't want to merge it, that's okay, but i detest adding more and more and more scope to this PR because if something breaks it's not as easily to disentangle in a revert PR.

Reply:
Because you are refactoring the mob to basic so part of that is refactoring the things it uses. This is unique to headslug so it makes no sense why this wouldn't be open to improving when you're refactoring the mob as a whole.

Reply:
because it would be a papered over fix for a framework that extends far beyond changeling eggs (e.g. this same framework is used in alien embryos). the absolute best fix for whatever you want would be in a dedicated PR that i could work on once i don't have three PRs open that actually leverages the backbone that it works on, instead of obscuring (a very nonexistent) problem.

Reply:
to be 100% clear - the majority of reviews you left on this PR were over this. I would much rather fix those four "wants" in a dedicated PR that ensures 1:1 parity by reworking the entire framework than just keep smushing more code to make jank look better just because it happened to surface up.

Reply:
This is surfacing up because it's a refactor of the mob as a whole, the jank I am talking about here is part of the headslug itself, it's their one ability that it uses, it's part of it's attacking.
Porting over to basic mobs was meant to help remove this jank-ass code and make shit more generic, not to simply repath and keep everything else the same.